From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: ponces <ponces26@gmail.com>
Date: Fri, 23 May 2026 00:00:00 +0000
Subject: [PATCH] Revert "Send KEYCODE_HOME for home button instead of
 startActivity directly"

This reverts commit 723d6e03a69cb2e50b8cc954a3ce571b674e858e. The
ISystemUiProxy AIDL in the GSI framework does not declare onKeyEvent.
---
diff --git a/quickstep/src/com/android/quickstep/OverviewCommandHelper.java b/quickstep/src/com/android/quickstep/OverviewCommandHelper.java
index b555bf7bef..56c9a00016 100644
--- a/quickstep/src/com/android/quickstep/OverviewCommandHelper.java
+++ b/quickstep/src/com/android/quickstep/OverviewCommandHelper.java
@@ -25,7 +25,6 @@ import android.content.Intent;
 import android.graphics.PointF;
 import android.os.SystemClock;
 import android.os.Trace;
-import android.view.KeyEvent;
 import android.view.View;
 
 import androidx.annotation.BinderThread;
@@ -71,7 +70,6 @@ public class OverviewCommandHelper {
     private final TouchInteractionService mService;
     private final OverviewComponentObserver mOverviewComponentObserver;
     private final TaskAnimationManager mTaskAnimationManager;
-    private final SystemUiProxy mSystemUiProxy;
     private final ArrayList<CommandInfo> mPendingCommands = new ArrayList<>();
 
     /**
@@ -91,12 +89,10 @@ public class OverviewCommandHelper {
 
     public OverviewCommandHelper(TouchInteractionService service,
             OverviewComponentObserver observer,
-            TaskAnimationManager taskAnimationManager,
-            SystemUiProxy systemUiProxy) {
+            TaskAnimationManager taskAnimationManager) {
         mService = service;
         mOverviewComponentObserver = observer;
         mTaskAnimationManager = taskAnimationManager;
-        mSystemUiProxy = systemUiProxy;
     }
 
     /**
@@ -234,7 +230,7 @@ public class OverviewCommandHelper {
                 case TYPE_HOME:
                     ActiveGestureLog.INSTANCE.addLog(
                             "OverviewCommandHelper.executeCommand(TYPE_HOME)");
-                    mSystemUiProxy.onKeyEvent(KeyEvent.KEYCODE_HOME);
+                    mService.startActivity(mOverviewComponentObserver.getHomeIntent());
                     return true;
                 case TYPE_SHOW:
                     // When Recents is not currently visible, the command's type is TYPE_SHOW
diff --git a/quickstep/src/com/android/quickstep/SystemUiProxy.java b/quickstep/src/com/android/quickstep/SystemUiProxy.java
index b5454cbd21..24db2edc78 100644
--- a/quickstep/src/com/android/quickstep/SystemUiProxy.java
+++ b/quickstep/src/com/android/quickstep/SystemUiProxy.java
@@ -210,17 +210,6 @@ public class SystemUiProxy implements ISystemUiProxy, NavHandle {
         }
     }
 
-    @Override
-    public void onKeyEvent(int keycode) {
-        if (mSystemUiProxy != null) {
-            try {
-                mSystemUiProxy.onKeyEvent(keycode);
-            } catch (RemoteException e) {
-                Log.w(TAG, "Failed call onKeyEvent", e);
-            }
-        }
-    }
-
     @Override
     public void onImeSwitcherPressed() {
         if (mSystemUiProxy != null) {
diff --git a/quickstep/src/com/android/quickstep/TouchInteractionService.java b/quickstep/src/com/android/quickstep/TouchInteractionService.java
index b088c017db..b43c5201ce 100644
--- a/quickstep/src/com/android/quickstep/TouchInteractionService.java
+++ b/quickstep/src/com/android/quickstep/TouchInteractionService.java
@@ -550,8 +550,7 @@ public class TouchInteractionService extends Service {
         mTaskAnimationManager = new TaskAnimationManager(this);
         mOverviewComponentObserver = new OverviewComponentObserver(this, mDeviceState);
         mOverviewCommandHelper = new OverviewCommandHelper(this,
-                mOverviewComponentObserver, mTaskAnimationManager,
-                SystemUiProxy.INSTANCE.get(this));
+                mOverviewComponentObserver, mTaskAnimationManager);
         mResetGestureInputConsumer = new ResetGestureInputConsumer(
                 mTaskAnimationManager, mTaskbarManager::getCurrentActivityContext);
         mInputConsumer.registerInputConsumer();
-- 
2.34.1
